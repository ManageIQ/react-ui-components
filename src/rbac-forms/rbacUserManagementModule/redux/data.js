export const data = {
  rows: [
    {
      id: 100,
      name: 'Adam the first',
      userid: 'Admin',
      email: 'email@neco.com',
      current_group: {
        label: 'Some group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Admin role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '06/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 101,
      name: 'Marge the middle',
      userid: 'La transparenza della qualita',
      email: 'email@jandak.com',
      current_group: {
        label: 'Mediocre group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Shot role',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '01/25/18 06:58:14 UTC',
      lastlogoff: '06/30/18 06:58:14 UTC',
    },
    {
      id: 102,
      name: 'Zlatko the last',
      userid: 'Peligro',
      email: 'email@jandak.com',
      current_group: {
        label: 'Admin group',
        onClick: () => console.log('Current group clicked'),
      },
      role: {
        label: 'Cloud-Users',
        onClick: () => console.log('Current role click'),
      },
      groups: [{
        label: 'Cloud-Operators',
        icon: 'group',
        groupId: 1,
        onClick: () => console.log('Group click'),
      }, {
        label: 'Cloud-Users',
        icon: 'group',
        groupId: 2,
        onClick: () => console.log('Group click'),
      }],
      lastlogon: '08/25/18 06:58:14 UTC',
      lastlogoff: '010/30/18 06:58:14 UTC',
    },
  ],
  columns: [{
    property: 'name',
    label: 'Full Name',
  }, {
    property: 'userid',
    label: 'Username',
  }, {
    property: 'email',
    label: 'E-mail',
  }, {
    property: 'current_group',
    label: 'Current Group',
  }, {
    property: 'role',
    label: 'Role',
  }, {
    property: 'lastlogon',
    label: 'Last Logon',
  }, {
    property: 'lastlogoff',
    label: 'Last Logoff',
  }],
  selectedUsers: null,
  groups: [
    {
      value: 1,
      label: 'Hope',
    },
    {
      value: 2,
      label: 'Christa',
    },
    {
      value: 3,
      label: 'Tania',
    },
    {
      value: 4,
      label: 'Mcintyre',
    },
    {
      value: 5,
      label: 'Tonya',
    },
    {
      value: 6,
      label: 'Marcie',
    },
    {
      value: 7,
      label: 'Cara',
    },
    {
      value: 8,
      label: 'Becker',
    },
    {
      value: 9,
      label: 'Foley',
    },
    {
      value: 10,
      label: 'Perkins',
    },
  ],
};
