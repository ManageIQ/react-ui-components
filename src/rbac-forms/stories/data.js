import { action } from '@storybook/addon-actions';

export const groups = [
  {
    value: 1,
    label: 'Hope',
  },
  {
    value: 2,
    label: 'Christa',
  },
  {
    value: 3,
    label: 'Tania',
  },
  {
    value: 4,
    label: 'Mcintyre',
  },
  {
    value: 5,
    label: 'Tonya',
  },
  {
    value: 6,
    label: 'Marcie',
  },
  {
    value: 7,
    label: 'Cara',
  },
  {
    value: 8,
    label: 'Becker',
  },
  {
    value: 9,
    label: 'Foley',
  },
  {
    value: 10,
    label: 'Perkins',
  },
];

export const editedUser = {
  name: 'Users full name',
  userid: 'admin jiny',
  chosen_group: [5, 7],
};

export const usersTableColumns = [{
  property: 'fullname',
  label: 'Full Name',
}, {
  property: 'username',
  label: 'Username',
}, {
  property: 'email',
  label: 'E-mail',
}, {
  property: 'currentgroup',
  label: 'Current Group',
}, {
  property: 'role',
  label: 'Role',
}, {
  property: 'lastlogon',
  label: 'Last Logon',
}, {
  property: 'lastlogoff',
  label: 'Last Logoff',
}];

export const usersTableRows = [
  {
    id: 100,
    fullname: 'Adam the first',
    username: 'Admin',
    email: 'email@neco.com',
    currentgroup: 'Some group',
    role: 'Admin role',
    lastlogon: '06/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 101,
    fullname: 'Marge the middle',
    username: 'La transparenza della qualita',
    email: 'email@jandak.com',
    currentgroup: 'Mediocre group',
    role: 'Shot role',
    lastlogon: '01/25/18 06:58:14 UTC',
    lastlogoff: '06/30/18 06:58:14 UTC',
  },
  {
    id: 102,
    fullname: 'Zlatko the last',
    username: 'Peligro',
    email: 'email@jandak.com',
    currentgroup: 'Super group',
    role: 'Shot role',
    lastlogon: '08/25/18 06:58:14 UTC',
    lastlogoff: '010/30/18 06:58:14 UTC',
  },
];

export const user = {
  name: 'Administrator',
  userid: 'Admin',
  email: 'mail@mail.com',
  current_group: {
    label: 'EvmGroup-super_administrator',
    onClick: () => action('Current group clicked'),
  },
  groups: [{
    label: 'Cloud-Operators',
    icon: 'group',
    groupId: 1,
    onClick: () => action('Group click'),
  }, {
    label: 'Cloud-Users',
    icon: 'group',
    groupId: 2,
    onClick: () => action('Group click'),
  }],
  role: {
    label: 'Cloud-Users',
    onClick: () => action('Current role click'),
  },
};
