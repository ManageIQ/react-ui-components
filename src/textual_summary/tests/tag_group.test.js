import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import TagGroup from '../tag_group';
import { tagGroupData } from '../data/tag_group';
import IconOrImage from '../icon_or_image';

describe('TagGroup', () => {
  /*
   *  has rows
   *    a) simple row
   *      has icon
   *      has value
   *    b) array row (multivalue)
   *      has icon
   *      values joined with "<b>&nbsp;|..."
   */
  it('renders just fine', () => {
    const image = shallow(<TagGroup items={tagGroupData.items} title={tagGroupData.title} />);
    expect(toJson(image)).toMatchSnapshot();
  });

  it('rendered array row with icon and values joined with "<b>&nbsp;|&nbsp;</b>")', () => {
    const tagData = {
      items: [
        {
          label: 'My Company X Tags',
          value: [
            { icon: 'fa fa-tag', label: 'Dan Test', value: ['Test 1'] },
            { icon: 'fa fa-tag', label: 'Demo bla', value: ['Policy', '2'] },
          ],
          hoverClass: 'no-hover',
        },
      ],
      title: 'Smart Management',
    };

    const wrapper = shallow(<TagGroup items={tagData.items} title={tagData.title} />);
    expect(wrapper.html()).toContain('Policy<b>&nbsp;|&nbsp;</b>2');
    expect(wrapper.containsMatchingElement(<IconOrImage icon="fa fa-tag" />)).toEqual(true);
  });
});
