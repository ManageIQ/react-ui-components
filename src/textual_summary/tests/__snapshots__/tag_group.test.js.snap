// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TagGroup renders just fine 1`] = `
<table
  className="table table-bordered table-striped table-summary-screen"
>
  <thead>
    <tr>
      <th
        align="left"
        colSpan="2"
      >
        Smart Management
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="0"
    >
      <td
        className="label"
      >
        Managed by Zone
      </td>
      <td>
        <IconOrImage
          icon="pficon pficon-zone"
          image={null}
          title={null}
        />
        default
      </td>
    </tr>
    <tr
      key="0"
    >
      <td
        className="label"
        rowSpan={2}
      >
        My Company X Tags
      </td>
      <td>
        <IconOrImage
          icon="fa fa-tag"
          image={null}
          title={null}
        />
        <span>
          Dan Test
          : 
          Test 1
        </span>
      </td>
    </tr>
    <tr
      key="1"
    >
      <td>
        <IconOrImage
          icon="fa fa-tag"
          image={null}
          title={null}
        />
        <span>
          Demo bla
          : 
          Policy
          <b>
             | 
          </b>
          2
        </span>
      </td>
    </tr>
  </tbody>
</table>
`;
